### PR TITLE
IGNITE-19783 StripedScheduledExecutorService is created for DistributionZoneManager#executor

### PR DESCRIPTION
**Pull request by sergeyuttsel**
https://issues.apache.org/jira/browse/IGNITE-19783

### CODE REVIEW

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java [hunk 1]
```diff
@@ -286,6 +287,7 @@ public DistributionZoneManager(
         nodesAttributes = new ConcurrentHashMap<>();
 
         executor = createZoneManagerExecutor(
+                Math.min(Runtime.getRuntime().availableProcessors() * 3, 20),
```
Comment (sergeyuttsel) [inline modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java:290]: This formula was used before a single thread executor was introduced.

File: modules/core/src/main/java/org/apache/ignite/internal/thread/StripedScheduledThreadPoolExecutor.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    public StripedScheduledThreadPoolExecutor(
+            int concurrentLvl,
+            ThreadFactory threadFactory,
+            RejectedExecutionHandler executionHandler
+    ) {
+        super(createExecutors(concurrentLvl, threadFactory, executionHandler));
+    }
+
+    private static ScheduledExecutorService[] createExecutors(
+            int concurrentLvl,
```
Comment (denis-chudov) [inline modules/core/src/main/java/org/apache/ignite/internal/thread/StripedScheduledThreadPoolExecutor.java:51]:
```suggestion
            int concurrencyLvl,
```

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZonesUtil.java [hunk 3]
```diff
@@ -571,17 +570,17 @@ public static Set<String> filterDataNodes(
...
-     * // TODO: IGNITE-19783 Need to use a striped executor.
      *
+     * @param concurrentLvl Number of threads.
      * @param namedThreadFactory Named thread factory.
      * @return Executor.
      */
-    static ScheduledExecutorService createZoneManagerExecutor(NamedThreadFactory namedThreadFactory) {
-        return new ScheduledThreadPoolExecutor(
-                1,
+    static StripedScheduledThreadPoolExecutor createZoneManagerExecutor(int concurrentLvl, NamedThreadFactory namedThreadFactory) {
```
Comment (denis-chudov) [inline modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZonesUtil.java:581]:
```suggestion
    static StripedScheduledThreadPoolExecutor createZoneManagerExecutor(int concurrencyLvl, NamedThreadFactory namedThreadFactory) {
```

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZonesSchedulersTest.java [hunk 4]
```diff
@@ -183,7 +194,8 @@ private static void testOrdering(BiConsumer<Long, Runnable> fn) throws Interrupt
                 if (counter.get() != j) {
                     sequentialOrder.set(false);
                 }
-            });
+            },
+                    STRIPE_0);
```
Comment (denis-chudov) [inline modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZonesSchedulersTest.java:198]: this needs correct formatting

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZonesSchedulersTest.java [hunk 5]
```diff
@@ -258,21 +270,24 @@ void testCancelScaleDownTasksOnStopTimers() {
...
+                        assertTrue(latch.await(3, TimeUnit.SECONDS));
+                    } catch (InterruptedException e) {
+                        throw new RuntimeException(e);
+                    }
+                },
+                STRIPE_0);
 
-        fn.accept(1L, () -> {});
+        fn.accept(1L, () -> {
+        }, STRIPE_1);
```
Comment (denis-chudov) [inline modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZonesSchedulersTest.java:290]: this also needs correct formatting, and the same cases below

File: modules/core/src/main/java/org/apache/ignite/internal/thread/StripedScheduledThreadPoolExecutor.java (hunk 2 above)
Comment (sergeyuttsel) [reply modules/core/src/main/java/org/apache/ignite/internal/thread/StripedScheduledThreadPoolExecutor.java]: Fixed.

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZonesSchedulersTest.java (hunk 4 above)
Comment (sergeyuttsel) [reply modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZonesSchedulersTest.java]: Thanks. Fixed.

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZonesSchedulersTest.java (hunk 5 above)
Comment (sergeyuttsel) [reply modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZonesSchedulersTest.java]: Fixed.

**Review by denis-chudov — APPROVED** [review]

**Pull request merged**